config/runtime.exs

```diff
@@ -82,109 +26,53 @@ end
 prod? = config_env() == :prod
 
 host =
-  case System.get_env("VIRTUAL_HOST") do
-    nil when prod? -> raise "Unset VIRTUAL_HOST environment variable"
+  case System.get_env("HOST") do
+    nil when prod? -> raise "Unset HOST environment variable"
     nil -> "localhost"
     other -> other
   end
 
 port =
-  case System.get_env("PORT") do
-    nil when prod? -> raise "Unset PORT environment variable"
-    nil -> 5002
-    other -> String.to_integer(other)
-  end
-
-jellyfish_address = System.get_env("JELLYFISH_ADDRESS") || "#{host}:#{port}"
-
-config :ex_dtls, impl: :nif
+  ConfigReader.read_port("PORT") ||
+    Application.get_env(:jellyfish, JellyfishWeb.Endpoint)[:http][:port]
 
 config :jellyfish,
-  webrtc_used: String.downcase(System.get_env("WEBRTC_USED", "true")) not in ["false", "f", "0"],
-  integrated_turn_ip:
-    System.get_env("INTEGRATED_TURN_IP", "127.0.0.1") |> ConfigParser.parse_integrated_turn_ip(),
-  integrated_turn_listen_ip:
-    System.get_env("INTEGRATED_TURN_LISTEN_IP", "127.0.0.1")
-    |> ConfigParser.parse_integrated_turn_ip(),
+  webrtc_used: ConfigReader.read_boolean("WEBRTC_USED") || true,
+  integrated_turn_ip: ConfigReader.read_ip("INTEGRATED_TURN_IP") || {127, 0, 0, 1},
+  integrated_turn_listen_ip: ConfigReader.read_ip("INTEGRATED_TURN_LISTEN_IP") || {127, 0, 0, 1},
   integrated_turn_port_range:
-    System.get_env("INTEGRATED_TURN_PORT_RANGE", "50000-59999")
-    |> ConfigParser.parse_integrated_turn_port_range(),
-  integrated_turn_tcp_port:
-    System.get_env("INTEGRATED_TURN_TCP_PORT")
-    |> ConfigParser.parse_port_number("INTEGRATED_TURN_TCP_PORT"),
+    ConfigReader.read_port_range("INTEGRATED_TURN_PORT_RANGE") || {50_000, 59_999},
+  integrated_turn_tcp_port: ConfigReader.read_port("INTEGRATED_TURN_TCP_PORT"),
   jwt_max_age: 24 * 3600,
   output_base_path: System.get_env("OUTPUT_BASE_PATH", "jellyfish_output") |> Path.expand(),
-  address: jellyfish_address
+  address: System.get_env("JELLYFISH_ADDRESS") || "#{host}:#{port}",
+  metrics_ip: ConfigReader.read_ip("METRICS_IP") || {127, 0, 0, 1},
+  metrics_port: ConfigReader.read_port("METRICS_PORT") || 9568
 
-config :opentelemetry, traces_exporter: :none
-
-if prod? do
-  token =
-    System.fetch_env!("SERVER_API_TOKEN") ||
-      raise """
-      environment variable SERVER_API_TOKEN is missing.
-      SERVER_API_TOKEN is used for HTTP requests and
-      server WebSocket authorization.
-      """
+config :jellyfish, JellyfishWeb.Endpoint,
+  secret_key_base:
+    System.get_env("SECRET_KEY_BASE") || Base.encode64(:crypto.strong_rand_bytes(48)),
+  http: [port: port]
```

Comment:
So current configuration works like this in each environment file we specify a port and then in runtime we check if we pass the variable `PORT` if yes we override a PORT value for Endpoint if no we set it up again to values from environment files?

Reply:
Yeap, exactly. We could do something like 

```elixir
if port = ConfigReader.read_port("PORT") do
  config :jellyfish, JellyfishWeb.Endpoint, http: [port: port]
 end
 ```
 but we also use port in the `jellyfish_address` so I reused code